AST-3158: pathways query (#17)
nckhell/awell-developers: content/api-reference/queries/get-pathways.mdx

---
title: Get pathways
description: Search pathways by status, type, patient and other criteria
---

> We suggest reading the page on our [domain model](/docs/getting-started/domain-model) first to get an understanding of the pathway model.

## Query

```graphql
query GetPathways($filters: FilterPathwaysGraphqlType) {
  pathways(filters: $filters) {
    success
    pathways {
      id
      title
      pathway_definition_id
      patient_id
      status
      status_explanation
      version # version number
      release_id # unique id representing the version
      start_date
      stop_date # if pathway is stopped
      complete_date # if pathway is completed
    }
  }
}
```

## Variables

### Filters

> All filter values are optional, if you don't provide any filter all pathways in your tenant will be returned.

Reviewer: ```suggestion
All filter values are optional, if you don't provide any filter all pathways in your organization will be returned.
```

All `in` filters accept a list of values and result in an `is any of` search criteria. The `eq` filter accepts a single string and results in `is exact match` search criteria.

> When specifying multiple filters, the resulting criteria is the conjuction of all filters.

Reviewer: ```suggestion
When specifying multiple filters, the resulting criteria is the conjunction of all filters.
```

```json
{
  "filters": {
    "pathway_definition_id": {
      "eq": ""
    },
    "patient_id": {
      "in": []
    },
    "release_id: { 
      "in": []
    },
    "start_date": {
      "gte": "",
      "lte": ""
    },
    "status": {
      "in": []
    },
    "version": {
      "in": []
    }
  }
}
```

### Pagination and sorting

> Pagination and sorting are highly recommended. Not using pagination means that the API will return all activities at once which will cause performance issues.

Reviewer: ```suggestion
Pagination and sorting are highly recommended. Not using pagination means that the API will return all pathways at once which will cause performance issues.
```

> Default sorting is based on the `start_date` field, sorted descendingly (i.e. newest to oldest).

Reviewer: ```suggestion
Default sorting is based on the `start_date` field, sorted from newest to oldest (i.e. descending).
```
